tsfc/coffee.py

```diff
@@ -257,7 +260,18 @@ def _expression_mathfunction(expr, parameters):
         'cyl_bessel_i': 'boost::math::cyl_bessel_i',
         'cyl_bessel_k': 'boost::math::cyl_bessel_k',
     }
-    name = name_map.get(expr.name, expr.name)
+    complex_name_map = {
+        'ln': 'clog',
+        'conj': 'conj'
+        # TODO: Are there different complex Bessel Functions?
```

Comment:
The standard bessel functions are defined over the field of complex numbers.  But boost for one does not actually implement them.

Reply:
The Boost functions are templated, but their implementation uses comparisons (e.g. `x / v < 0.25`) which does not compile when the templates are instantiated with `std::complex<double>`.